builder/transaction_builder.go

```diff
@@ -111,6 +124,137 @@ func (b *TransactionBuilder) AddTaggedDataPayload(payload *iotago.TaggedData) *T
+			if accountBalance < storedMana {
+				// not enough mana for this account on the input side
+				// => set the remaining account bound mana for this account to 0
+				accountBoundManaInputs[accountID] = 0
+
+				// subtract the remainder from the unbound mana
+				return updateUnboundManaBalance(storedMana - accountBalance)
+			}
+
+			// there is enough account bound mana for this account, subtract it from there
+			accountBoundManaInputs[accountID] -= storedMana
+
+			return nil
+		}
+
+		// no account bound mana available for the given account, subtract it from the unbounded mana
+		return updateUnboundManaBalance(storedMana)
+	}
+
+	if storedManaOutputIndex >= len(b.transaction.Outputs) {
+		return setBuildError(ierrors.Errorf("given storedManaOutputIndex does not exist: %d", storedManaOutputIndex))
+	}
+
+	// subtract the stored mana on the outputs side
+	for _, o := range b.transaction.Outputs {
+		switch output := o.(type) {
+		case *iotago.AccountOutput:
+			// mana on account outputs is locked to this account
+			if err := updateUnboundAndAccountBoundManaBalances(output.AccountID, output.StoredMana()); err != nil {
+				return setBuildError(ierrors.Wrap(err, "failed to subtract the stored mana on the outputs side"))
+			}
+
+		default:
+			// check if the output locked mana to a certain account
+			if accountID, isManaLocked := hasManalockCondition(output); isManaLocked {
+				if err := updateUnboundAndAccountBoundManaBalances(accountID, output.StoredMana()); err != nil {
+					return setBuildError(ierrors.Wrap(err, "failed to subtract the stored mana on the outputs side"))
+				}
+			} else {
+				if err := updateUnboundManaBalance(output.StoredMana()); err != nil {
+					return setBuildError(ierrors.Wrap(err, "failed to subtract the stored mana on the outputs side"))
+				}
+			}
+		}
+	}
+
+	// subtract the already alloted mana
+	for _, allotment := range b.transaction.Allotments {
+		if err := updateUnboundAndAccountBoundManaBalances(allotment.AccountID, allotment.Value); err != nil {
+			return setBuildError(ierrors.Wrap(err, "failed to subtract the already alloted mana"))
+		}
+	}
+
+	// calculate the minimum required mana to issue the block
+	minRequiredMana, err := b.MinRequiredAllotedMana(b.api.ProtocolParameters().WorkScoreStructure(), rmc, blockIssuerAccountID)
+	if err != nil {
+		return setBuildError(ierrors.Wrap(err, "failed to calculate the minimum required mana to issue the block"))
+	}
+
+	// subtract the minimum required mana to issue the block
+	if err := updateUnboundAndAccountBoundManaBalances(blockIssuerAccountID, minRequiredMana); err != nil {
+		return setBuildError(ierrors.Wrap(err, "failed to subtract the minimum required mana to issue the block"))
+	}
+
+	// allot the mana to the block issuer account (we increase the value, so we don't interfere with the already alloted value)
+	b.IncreaseAllotment(blockIssuerAccountID, minRequiredMana)
```

Comment:
Should we check if there is already allotment to this accountID and increase only up to the minRequiredMana rather than adding minRequiredMana extra?

Reply:
I think its better to increase the allotment, as written in the comment.